### ISSUE TITLE
Parentheses in full path name prevent loading notes on startup

### ISSUE THREAD

**Issue by ashtuchkin** [foambubble/foam#923]
- Foam version: v0.17.3
- Platform: Mac
- Issue occur on the [foam template](https://github.com/foambubble/foam-template) repo: Yes

**Summary**
When full path to repository contains parenthesis (e.g. `My folder (1)`), foam does not load the notes on start, which prevents link recognition, graph is empty, etc.

My best guess why that happens is that you use [micromatch library](https://www.npmjs.com/package/micromatch) here: https://github.com/foambubble/foam/blob/17cb61948047d53227ebd9eebd1c16472f7c5400/packages/foam-vscode/src/core/services/datastore.ts#L79-L87

.. and micromatch treats parentheses characters (`(` and `)`) in include globs and exclude globs specially: see [extglobs](https://www.npmjs.com/package/micromatch#extglobs). See logs below. So you probably need to escape them (and all other special characters) before using.

Why it's important? This bug prevents putting the foam folder into a Dropbox account that is paired with another Dropbox account. In this case, personal dropbox folder is named `Dropbox (Personal)` and there's no way to rename it.

**Steps to reproduce**
1. 
```
mkdir "Test (1)"
cd "Test (1)"
git clone git@github.com:foambubble/foam-template.git
```
2. Open VS Code in that folder.
3. Observe no notes in graph.
4. Observe "[info - 3:48:29 PM] Loaded 0 notes" in the foam logs.

**Additional information**
Full log:
```
Foam Logging: info
[info - 3:48:28 PM] Starting Foam
[info - 3:48:28 PM] Workspace folders: 
[
  "/Users/ashtuchkin/Test (1)/foam-template"
]
[info - 3:48:28 PM] Glob patterns
{
  "includeGlobs": [
    "/Users/ashtuchkin/Test (1)/foam-template/**/*"
  ],
  "ignoreGlobs": [
    "/Users/ashtuchkin/Test (1)/foam-template/**/.vscode/**/*",
    "/Users/ashtuchkin/Test (1)/foam-template/**/_layouts/**/*",
    "/Users/ashtuchkin/Test (1)/foam-template/**/_site/**/*",
    "/Users/ashtuchkin/Test (1)/foam-template/**/node_modules/**/*",
    "/Users/ashtuchkin/Test (1)/foam-template/**/.git",
    "/Users/ashtuchkin/Test (1)/foam-template/**/.svn",
    "/Users/ashtuchkin/Test (1)/foam-template/**/.hg",
    "/Users/ashtuchkin/Test (1)/foam-template/**/CVS",
    "/Users/ashtuchkin/Test (1)/foam-template/**/.DS_Store",
    "/Users/ashtuchkin/Test (1)/foam-template/**/Thumbs.db",
    "/Users/ashtuchkin/Test (1)/foam-template/**/node_modules"
  ]
}
[info - 3:48:29 PM] Loaded 0 notes
```

**Comment by Am3ra** [comment]
So would using the `noextglob: true` option fix this?

**Comment by ashtuchkin** [comment]
I guess; someone would need to test it works.

**Comment by riccardoferretti** [comment]
I checked and the setting didn't make a difference

**Comment by riccardoferretti** [comment]
This is now fixed as of 0.20.2
